dev_guide/managing_images.adoc

```diff
@@ -0,0 +1,690 @@
+
+----
+$ oc secrets new <pull_secret_name> .dockercfg=<path/to/.dockercfg>
+----
+
+If you do not already have a *_.dockercfg_* file for the secured registry, you
+can create a secret by running:
+
+----
+$ oc secrets new-dockercfg <pull_secret_name> \
+    --docker-server=<registry_server> --docker-username=<user_name> \
+    --docker-password=<password> --docker-email=<email>
+----
+
+To use a secret for pulling images for pods, you must add the secret to your
+service account. The name of the service account in this example should match
+the name of the service account the pod will use; *default* is the default
+service account:
+
+----
+$ oc secrets add serviceaccount/default secrets/<pull_secret_name> --for=pull
+----
+
+To use a secret for pushing and pulling build images, the secret must be
+mountable inside of a pod. You can do this by running:
+
+----
+$ oc secrets add serviceaccount/builder secrets/<pull_secret_name>
+----
+
+[[creating-an-image-stream-by-manually-pushing-an-image]]
+== Creating an Image Stream by Manually Pushing an Image
+
+An image stream can be automatically created by manually pushing an image to the
+internal registry. This is only possible using {product-title}'s internal
+registry.
+
+To perform the following procedure, the destination project you push to must
+already exist and the user must be authorized to `{get, update}
+"imagestream/layers"` in that project. The *system:image-pusher* role can
+provide these permissions. If you are a project administrator, then you would
+also have these permissions.
+
+. First, link:#accessing-the-internal-registry[log in to the internal registry].
+
+. Then, tag your image using the appropriate internal registry location. For
+example, if you had already pulled the *docker.io/centos:centos7* image locally:
++
+====
+----
+$ docker tag docker.io/centos:centos7 172.30.48.125:5000/test/my-image
+----
+====
+
+. Finally, push the image to your internal registry. For example:
++
+====
+----
+$ docker push 172.30.48.125:5000/test/my-image
+The push refers to a repository [172.30.48.125:5000/test/my-image] (len: 1)
+c8a648134623: Pushed
+2bf4902415e3: Pushed
+latest: digest: sha256:be8bc4068b2f60cf274fc216e4caba6aa845fff5fa29139e6e7497bb57e48d67 size: 6273
+----
+====
+
+. Verify that the image stream was created:
++
+====
+----
+$ oc get is
+NAME       DOCKER REPO                        TAGS      UPDATED
+my-image   172.30.48.125:5000/test/my-image   latest    3 seconds ago
+----
+====
+
+[[importing-tag-and-image-metadata]]
+== Importing Tag and Image Metadata
+
+An image stream can be configured to import tag and image metadata from an image
+repository in an external Docker image registry. To create an image stream from
+an external registry, set the `*spec.dockerImageRepository*` field
+appropriately. For example:
+
+====
+----
+apiVersion: "v1"
+kind: "ImageStream"
+metadata:
+  name: "ruby"
+spec:
+  dockerImageRepository: "openshift/ruby-20-centos7"
+----
+====
```

Comment:
Yes. Generally you can either create the IS manually or using `oc import-image --from=<here goes docker pull spec>`. So id would be nice to separate the two. Since after reading it user might get the impression that first he needs to create and then import, which is not true. Import will happen automatically when IS is created (or rather shortly after). `oc import-image` can be used instead of that or to re-import an image.